lib/Player.rb

```diff
@@ -0,0 +1,34 @@
+class Player
+    attr_accessor :name, :piece
+    def initialize(name, piece, board)
+        @name = name 
+        @piece = piece
+        @board = board
+    end
+
+    def ask_move
```

Comment:
Is there a way to separate UI-related code (which in this case is the logic asking for a user's input) from the player class? Perhaps a module dedicated to UI logic will do the job and you can use the output of that in this class to use as the player's move/chosen square